src/main/scala/is/hail/methods/LinearRegressionBurden.scala

```diff
@@ -0,0 +1,108 @@
+package is.hail.methods
+
+import breeze.linalg._
+import is.hail.annotations.Annotation
+import is.hail.expr._
+import is.hail.keytable.KeyTable
+import is.hail.stats._
+import is.hail.utils._
+import is.hail.variant._
+import net.sourceforge.jdistlib.T
+import org.apache.spark.sql.Row
+
+object LinearRegressionBurden {
+
+  def apply(vds: VariantDataset, keyName: String, variantKeySetVA: String, aggregateExpr: String, ySA: String, covSA: Array[String], dropSamples: Boolean): KeyTable = {
+
+    val (y, cov, completeSamples) = RegressionUtils.getPhenoCovCompleteSamples(vds, ySA, covSA)
+
+    val n = y.size
+    val k = cov.cols
+    val d = n - k - 1
+
+    if (d < 1)
+      fatal(s"$n samples and $k ${ plural(k, "covariate") } including intercept implies $d degrees of freedom.")
+
+    info(s"Running linreg_burden, aggregated by key $keyName using $aggregateExpr, on $n samples with $k ${ plural(k, "covariate") } including intercept...")
+
+    val completeSamplesSet = completeSamples.toSet
+
+    if (completeSamplesSet(keyName))
+      fatal(s"Sample name conflicts with the key name $keyName")
+
+    if (!dropSamples) {
+      val conflicts = completeSamplesSet.intersect(LinearRegression.schema.fields.map(x => x.name).toSet)
+      if (conflicts.nonEmpty)
+        fatal(s"Sample names conflict with these reserved statistical names: ${conflicts.mkString(", ")}")
+    }
+
+    val (variantKeysType, variantKeysQuerier) = vds.queryVA(variantKeySetVA)
+
+    if (variantKeysType != TSet(TString))
+      fatal(s"Variant keys must be of type Set(String), got $variantKeysType")
+
+    val pattern = """gs.map\(g => (.*)\).(.*)""".r
```

Comment:
We are not parsing by regular expression.

The simplest thing is to just set `genotypeExpr` to `g` in the `makeKT` call.

Then, have the user write an expression that has `gs` in scope, call that `aggregateWith`, and write this:
```
val aggExpr = completeSamples.map(s => s"`$s` = let gs = `$s` in $aggregateWith").mkString(", ")
```

But actually don't like that either, it's way too easy for them to fuck up and include commas and get completely insane error messages. But I don't see a better option right now.

What we actually need is some way to parse the value side of a named expression independently of the name side.